lib/bundler/explain/parser.rb

```diff
@@ -1,26 +1,34 @@
 module Bundler
   module Explain
     class Parser
-      def initialize(file)
-        @file = file
-        @tree = {}
+      attr_reader :dependencies, :from_gemfile, :errors
+
+      def initialize(params = {})
+        @gemfile = params[:gemfile]
+        @gemfile_lock = params[:gemfile_lock]
```

Comment:
Keyword arguments.

Reply:
BTW, I believe that ONLY `Gemfile.lock` is necessary (in any case, they are always in the same folder and with predictable names), so having two params is over-engineering a bit.